### DISCUSSION TITLE
Flux v2 Terraform Provider

### DISCUSSION THREAD

**Discussion by stefanprodan** [fluxcd/flux2#398] [Proposals] [answered]
## Context

Many Flux v1 users are installing Flux and Helm Operator using the Helm Terraform provider as part of their cluster bootstrap procedure. Since Helm v3 doesn't handle CRDs updates, we can't offer a chart for Flux v2 that would reliably manage the lifecycle of the GitOps Toolkit CRDs.

## Proposal

The Flux CLI `install` and `bootstrap` commands are using the `flux2/pkg/manifestgen` Go module to download the GitOps Toolkit manifests (CRDs, controllers, RBAC, network polices, etc) from GitHub releases and customise them according to the user's supplied args.

We could create a Terraform provider that imports the `manifestgen` package. The provider will install Flux v2 on Kubernetes clusters and bootstrap a Git repository in the same way `flux bootstrap` does it. The provider should expose all the install options available in the CLI and use the same defaults as `flux install/bootstrap`.

The provider will live at `fluxcd/terraform-provider-flux` and will be published on the official Terraform Registry.

**Comment by squaremo** [comment]
:+1:

I'm fine with it either way, but I recall that in the meeting the idea was to include it in the flux2 repo -- can we record here the reason that a separate repo was settled on, please.

**Reply by stefanprodan** [reply]
The repo must follow the Terraform Registry convention `terraform-provider-<project>`, that's why we decided to go with a dedicated repo and use the defaults from `manifestgen` package, this way we don't duplicate the CLI defaults.

**Comment by hiddeco** [comment]
:+1:

**Comment by phillebaba** [comment, marked as answer]
Just so that the information is here if someone comes looking for the provider, here is a link to the repository.

https://github.com/fluxcd/terraform-provider-flux
